### PR TITLE
More editorial fixes

### PR DESCRIPTION
**Pull request by cabo**
Daniel Parker's comments
Fix ABNF (OK, that is not editorial) and dehexify

### CODE REVIEW

File: draft-normington-jsonpath-latest.md
Comment (glyn) [inline draft-normington-jsonpath-latest.md:256]: Is "the argument" well defined at this point in the document? It's slightly ambiguous since there are two inputs: the JSONPath itself and the JSON value.

Also, I suggest adding a comma, thus: "Since a JSON data item is anonymous, JSONPath uses...".

(Sorry I can't use a github suggestion here as I'm reviewing the commits separately.)

File: draft-normington-jsonpath-latest.md
Comment (glyn) [inline draft-normington-jsonpath-latest.md:4]: Should be -latest, for consistency.

**Review by glyn — CHANGES_REQUESTED** [review]
Please address my comments and delete -00.md, update the script/README as necessary for the renaming back to -latest, and regenerate.

File: draft-normington-jsonpath-latest.md
Comment (cabo) [reply draft-normington-jsonpath-latest.md]: There is a definition for "argument".  Of course, we could introduce this once more in the text, but maybe not here.
(And if you have a better name for the JSON data item under consideration by the JSONPath query, I'm all ears.)

Comment (glyn) [reply draft-normington-jsonpath-latest.md]: Oops, missed that definition for some reason. Thanks.

**Comment by glyn** [comment]
I rewrote history after checking with @cabo. HEAD now points at the commit _before_ I renamed to -00. (The renaming is tagged as draft-normington-jsonpath-00.) So this PR should look like a set of changes on the *-latest.* files.

File: scripts/gen.sh
Comment (glyn) [inline scripts/gen.sh:14]: This needs reworking to match the new naming scheme `draft-normington-jsonpath.*`.

**Comment by glyn** [comment]
@cabo The README also needs changing to reflect the new naming scheme `draft-normington-jsonpath.*`.

**Review by glyn — COMMENTED** [review]
My changes have been or are being addressed. Since I'm away until 16 December, I'm going to abstain and leave it to others to approve/merge. Otherwise, I'll pick this up when I'm back.

File: draft-normington-jsonpath.md
Comment (gregsdennis) [inline draft-normington-jsonpath.md:179]: Perhaps we should define "key" and even declare "property" as a synonym.

File: draft-normington-jsonpath.md
Comment (gregsdennis) [inline draft-normington-jsonpath.md:277]: Do we want to avoid the use of "expressions" here so that it's not confused with query expressions (defined on line 290)?

File: draft-normington-jsonpath.md
Comment (gregsdennis) [inline draft-normington-jsonpath.md:297]: "element" instead of "item" to align with the definition.

File: draft-normington-jsonpath.md
Comment (gregsdennis) [inline draft-normington-jsonpath.md:308]: Why do we need "item"?  "Element" is strictly defined.  Maybe just use that.

File: draft-normington-jsonpath.md
Comment (gregsdennis) [inline draft-normington-jsonpath.md:414]: Why are we specifying JSON for objects everywhere?  What other meaning does "object" have that warrants this specificity?

**Review by gregsdennis — CHANGES_REQUESTED** [review]

File: draft-normington-jsonpath.md
Comment (cabo) [reply draft-normington-jsonpath.md:179]: Property is a synonym for member, as is entry and field.
How many of these synonyms do we need?
(RFC 8259 calls the keys "names", which is not better, except maybe in combinations such as "field name" or "member name".)

Comment (gregsdennis) [reply draft-normington-jsonpath.md:179]: "key" and "property" are by far the most common terms I see in use.

Comment (cabo) [reply draft-normington-jsonpath.md:179]: And I've been caught calling keys "labels".  Some people like to call them "tags".
My first hit when searching was https://www.w3schools.com/Js/js_json_objects.asp, which you may like or not like (they do confuse JSON objects with their representation), but for them introducing JSON objects as key/value pairs seemed obvious, not needing further explanation.

File: scripts/gen.sh
Comment (cabo) [reply scripts/gen.sh:14]: Done in e28d965

File: draft-normington-jsonpath.md
Comment (mkmik) [reply draft-normington-jsonpath.md:414]: I agree. Also other primitives like number and string are also _JSON_ numbers and _JSON_ strings, (with all the subtleties that come from that, both from spec and underlying json engine implementation choices allowed by the JSON spec)

File: draft-normington-jsonpath.md
Comment (mkmik) [reply draft-normington-jsonpath.md:179]: Key/value is indeed very common. "field" is less common but nevertheless attested in the wild.

I tend to prefer `field name/value` mainly for two reasons: a) it builds on top of RFC 8259 terminology b) the "field" qualifier makes sense also when referring to the value (the _field value_)

File: scripts/gen.sh
Comment (glyn) [reply scripts/gen.sh:14]: Thanks. For the record, I marginally prefer the explicit `-latest` suffix as it looks like a placeholder waiting to be replaced by a version number. But I can certainly live with this alternative.

File: draft-normington-jsonpath.md
Comment (glyn) [inline draft-normington-jsonpath.md:255]: The anchor `#overview` clashes with the default anchor for the _other_ `Overview` section.

File: draft-normington-jsonpath.md
Comment (glyn) [inline draft-normington-jsonpath.md:202]: Normalized form is not yet well defined. Is it intended to be [canonical](https://en.wikipedia.org/wiki/Canonical_form) too or simply normalized (but not necessarily unique among normalized alternatives)?

File: draft-normington-jsonpath.md
Comment (danielaparker) [reply draft-normington-jsonpath.md:179]: In the spirit of "you can't please everyone", my own opinion from an implementer's viewpoint is that use of synonyms is best avoided. Implementer's want a parsimonious vocabulary with precise definitions, preferably one that's consistent with the core specifications (RFC 8259), the introduction of synonyms which are "suggestive" only frightens them. Words mean different things in different contexts, for example, "field" in the Java specifications refers to a type and an identifier, in a CODASYL data record it means something else. "property" isn't mentioned in RFC 8259, why introduce it here?      

Regarding "perhaps we should define 'key'", if it appears in the ABNF, it is defined.

Comment (cabo) [reply draft-normington-jsonpath.md:174]: It probably would help to keep "name" around for names in jsonpath.
But given that "key" is not defined in 8259, I'll stick with "name" for now, expanding to "member name" when needed.

File: draft-normington-jsonpath.md
Comment (cabo) [reply draft-normington-jsonpath.md:276]: This is one of many places that uses "jsonpath expression".  We could all rename those into "query".  Holding off with that right now.

File: draft-normington-jsonpath.md
Comment (cabo) [reply draft-normington-jsonpath.md:307]: The slash is XML/JSON.  Element on the JSON side means something different than element on the XML side.

File: draft-normington-jsonpath.md
Comment (cabo) [reply draft-normington-jsonpath.md:413]: I changed them all to object.  I continue to believe this is bad usage, but if the spec is mainly directed at people who work exclusively with JSON, that may be a simplification.

File: draft-normington-jsonpath.md
Comment (cabo) [reply draft-normington-jsonpath.md:197]: No, that's why it points to a section that explains in which way this is normalized.

File: draft-normington-jsonpath.md
Comment (gregsdennis) [reply draft-normington-jsonpath.md]: Maybe that distinction could be made clearer (later).

File: draft-normington-jsonpath.md
Comment (gregsdennis) [reply draft-normington-jsonpath.md]: My feeling is that this is a JSON-related specification, so it's likely to be assumed.  We _could_ add a statement in the declarations at the beginning to the effect of, "All usages of 'object', etc, mean the JSON usage unless otherwise specified."

**Review by gregsdennis — APPROVED** [review]
I'm happy with this as an intermediate state to be merged so that we don't block further development.

**Review by glyn — APPROVED** [review]
Yes, let's merge this to allow forward progress to resume.

**Pull request merged**
